profiles/mercury.raml

```diff
@@ -159,4 +160,14 @@ validations:
         pattern: ^v[0-9]+$
         minCount: 1
 
+  base-uri-matches-pattern:
+    message: The baseUri must match the pattern - https://{shortCode}.api.commercecloud.salesforce.com/<api-family>/<api-name>/{version}
+    targetClass: apiContract.WebAPI
+    functionConstraint:
+      code: |
+        function(resource) {
+          const baseUriRegex = /^https:\/\/{shortCode}.api.commercecloud.salesforce.com\/[a-z]+-?[a-z]+\/[a-z]+-?[a-z]+\/{version}$/;
+          const server = resource['apiContract:server'];
+          return server ? baseUriRegex.test(server[0]['core:urlTemplate'][0]) : false;
```

Comment:
server[0] seems a bit easy to break.  Can you have multiple servers specified?  Not sure we want to do this with custom code.